code/Controller/AssetAdminOpen.php

```diff
@@ -411,9 +411,19 @@ private function filterFiles(DataList $list, array $filter): DataList
             $list = $list->filter("Created:LessThanOrEqual", $toDate->dataValue().' 23:59:59');
             $search = true;
         }
-        // Categories (mapped to extensions through the enum type automatically)
+        // Categories (mapped to extensions)
         if (!empty($filter['appCategory'])) {
-            $list = $list->filter('Name:EndsWith', $filter['appCategory']);
+            $categoryValues = File::config()->get('app_categories');
+            // Sanitise category names (some contain slashes) and make all upper case
+            foreach ($categoryValues as $key => $extensions) {
+                unset($categoryValues[$key]);
+                $newKey = strtoupper(preg_replace('/[^[[:alnum:]]]*/', '', $key));
+                $categoryValues[$newKey] = $extensions;
+            }
+            if (!isset($categoryValues[$filter['appCategory']])) {
```

Comment:
```suggestion
            $appCategory = $filter['appCategory'];
            if (!isset($categoryValues[$appCategory])) {
```
Worth expanding out to make it easier to read

Reply:
I don't think it is worth it